src/patronus/evals/evaluators.py

```diff
@@ -0,0 +1,252 @@
+import typing
+from typing import Any, Optional, Union
+
+from opentelemetry.trace import get_current_span
+
+from patronus import api_types
+from patronus.evals.exporter import get_exporter
+from patronus.evals.types import EvaluationResult
+from patronus.tracing.attributes import Attributes
+from patronus.tracing.decorators import traced
+from patronus.tracing.logger import get_logger, get_patronus_logger
+
+log = logging.getLogger("patronus.core")
+
+
+def create_field_sanitizer(pattern: str, max_len: int, replace_with: str):
+    def sanitize(value: str) -> str:
+        return re.sub(pattern, replace_with, value[:max_len])
+
+    return sanitize
+
+
+sanitize_project_name = create_field_sanitizer(r"[^a-zA-Z0-9_ -]", max_len=50, replace_with="_")
+sanitize_app = create_field_sanitizer(r"[^a-zA-Z0-9-_./ -]", max_len=50, replace_with="_")
+
+
+def handle_eval_output(
+    *,
+    class_name: Optional[str],
+    fn_name: str,
+    evaluator_id: str,
+    metric_name: str,
+    metric_description: str,
+    fn_sign: inspect.Signature,
+    args,
+    kwargs,
+    ret_value: typing.Any,
+    duration: datetime.timedelta,
+):
+    if ret_value is None:
+        return
+
+    ev: EvaluationResult = coerce_eval_output_type(ret_value, class_name, fn_name)
+    pat_logger = get_patronus_logger()
+    span_context = get_current_span().get_span_context()
+    bound_args = fn_sign.bind(*args, **kwargs)
+    log_id = pat_logger.log(
+        body={"evaluation_data": {**bound_args.arguments, **bound_args.kwargs}},
+    )
+
+    # TODO there should be a better way of extracting project name and an app from execution context
+    project_name = pat_logger._instrumentation_scope.attributes.get(Attributes.project_name)
+    if project_name is not None:
+        project_name = sanitize_project_name(project_name)
+    app = pat_logger._instrumentation_scope.attributes.get(Attributes.app)
+    if app is not None:
+        app = sanitize_app(app)
+
+    try:
+        eval_payload = api_types.ClientEvaluation(
+            log_id=log_id,
+            project_name=project_name,
+            app=app,
+            experiment_id=pat_logger._instrumentation_scope.attributes.get(Attributes.experiment_id),
+            evaluator_id=evaluator_id,
+            criteria=None,  # TODO
+            pass_=ev.pass_,
+            score=ev.score,
+            text_output=ev.text_output,
+            metadata=ev.metadata,
+            explanation=ev.explanation,
+            explanation_duration=ev.explanation_duration,
+            evaluation_duration=ev.evaluation_duration or duration,
+            metric_name=metric_name,
+            metric_description=metric_description,
+            dataset_id=None,  # TODO
+            dataset_sample_id=None,  # TODO
+            created_at=datetime.datetime.now(datetime.timezone.utc),
+            tags=ev.tags,
+            trace_id=hex(span_context.trace_id)[2:].zfill(32),
+            span_id=hex(span_context.span_id)[2:].zfill(16),
```

Comment:
Are you handling empty trace_id, span_id? By default it's set to 0x000000 if not given - just make sure it works